pandas/io/common.py

```diff
@@ -179,6 +179,7 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
     filepath_or_buffer : a url, filepath (str, py.path.local or pathlib.Path),
                          or buffer
     encoding : the encoding to use to decode py3 bytes, default is 'utf-8'
+    mode : {'rb', 'wb', 'ab'}
```

Comment:
why do we need this here at all? your are already hardcoding it as needed in the ``pandas/io/s3.py`` anyhow.

Reply:
`get_filepath_or_buffer` is called for both read and write.

Reply:
@jberka for more context see here https://github.com/pandas-dev/pandas/pull/19135#discussion_r160474151

Reply:
I /think/ that I need it in common.py because parquet.py imports `from pandas.io.common import get_filepath_or_buffer` 

I've made an attempt to pass 'None' as default, and it overwrote the default 'rb' value that I've set. I can't set it by default to 'rw' because of the linked comment discussion.

Having said that, python is great but i don't consider myself a pro in anything. I'd be happy to merge a commit with a different approach into this PR.

Reply:
you can set ``mode=None`` for the default, then in the body, if its ``None`` you can  ``mode='rb'`` for reading and ``'wb'`` for writing.

Reply:
@jreback Would a default of `mode=None` be a better approach then `mode='rb'`? 

A solution where we don't add `mode=` to the signature of `get_filepath_or_buffer` is welcomed, but if it's `mode=None` or `mode='rb'` then I tend to favor `'rb'` as it hints on the default.

Reply:
yes mode=None is better

Reply:
So is it a clash of coding styles? https://github.com/dask/s3fs/blob/master/s3fs/core.py#L263 

Busy doing learning here.. would be glad to know what's a better approach and what is a convention in python.